### PR TITLE
Keyboard shortcuts and fuzzy search

### LINKED ISSUE #59: Keyboard Shortcuts
It would be great we can create a set of keyboard shortcuts. Then we can do following this very easily:

* toggle controls (the menu on the left)
* toggle action logger
* full screen component preview window

### PR DESCRIPTION
**Pull request by ritz078**
Need to add some tests.

Current key bindings:
`Ctrl + Shift + O` => Open search box
`Ctrl + Shift + L` => Toggle left panel
`Ctrl + Shift + B` => Toggle bottom panel
`Ctrl + Shift + F` => Toggle Full screen view

Closes #59 and #121

### CODE REVIEW

**Comment by arunoda** [comment]
Okay. This is super. 
One just UI change. Can you keyboard icon next to the logo and show the keymap.

**Comment by arunoda** [comment]
Shall we keep the current filter box as it is.

File: src/client/ui/controls.js [hunk 1]
```diff
@@ -194,4 +217,5 @@ StorybookControls.propTypes = {
   selectedStory: React.PropTypes.string,
   onKind: React.PropTypes.func,
   onStory: React.PropTypes.func,
+  syncedStore: React.PropTypes.object,
```
Comment (arunoda) [inline src/client/ui/controls.js]: Hey, is there anyway we can use a function rather not passing the syncedStore.

**Comment by arunoda** [comment]
@mnmtanish Could you go on this as well. 
It's always better if we some fresh eyes at this.

**Comment by arunoda** [comment]
A few ideas:

* Shall we use 'CMD'  + 'SHIFT' in mac. Just like in Atom, ST2/3
* Shall we use 'CMD + SHIFT + P' for lookup just like we used to do in Editors
* In the full screen mode, still there's a margin around the preview. I hope we can remove that.

I think, we can break this PR in multiple and then we can easily review this and pull into core. What do you think of dividing this PR into three:

1. Basic Keyboard Support (the logistics)
2. Full screen support + other minor keyboard command
3. Search box

Then we can review individual components and take them in quicker.

@ritz078 What do you think?

File: src/client/ui/controls.js (hunk 1 above)
Comment (ritz078) [inline src/client/ui/controls.js]: Yes , we can. will do it.

**Comment by ritz078** [comment]
I will definitely add `CMD` support. And regarding separate PRs each functionality is dependent on other. If i remove the full screen support , it will just remove 10 lines from code. So i will suggest it is kept this way. I still have refactorings and improvements to do. So I have already written `WIP` in the title. Keep your ideas coming.

**Comment by ritz078** [comment]
There's no point in keeping a separate search box which can't search through stories. The two search boxes will differ in behaviour and that's not good. Even the editors don't have separate search bars because on small screens it takes considerable space. But I may be wrong.

**Comment by arunoda** [comment]
Okay. Let's remove other shortcuts from this as well. 
Just work on the search.  

Having the filter box is pretty important. Here's why:

We have a lot of stories (and a lot of kinds) in one of our app. 
We need to use the left navigation panel. We are not looking for a jump to story functionality. 
That's why we need the filter box.

**Comment by ritz078** [comment]
I have already fixed all the other shortcuts so why remove them now ?

**Comment by arunoda** [comment]
Ah okay. Then fine.

**Comment by ritz078** [comment]
You can even search for a kind and the results of that kind will come at a top with the name of the kind written to the right. Eventually, you land at a story anyways. Till now we were landing at the first story by default. There's nothing like a separate page for kind.

**Comment by arunoda** [comment]
Yes. true. 
But that's not the case. 
We don't need to search for every time we need to switch stories. 
Just want to browse through the left panal. 

Sometimes it's not just a single kind but some similar.

**Comment by ritz078** [comment]
Cool. I will restore the filter panel.

**Comment by arunoda** [comment]
Let's keep the filter as it is. 
If we think the jump to story is better after sometimes using it, let's remove the filter. 
For now, let's keep it.

**Comment by arunoda** [comment]
Ha ha. You reply faster. 
Cool. Thanks.

**Comment by ritz078** [comment]
A suggestion for future : We can keep both kind and story filter. If you add a prefix like `@` in search , you only filter the kinds and by default stories.

**Comment by arunoda** [comment]
@ritz078 That's good.

**Comment by ritz078** [comment]
@arunoda have a look now.

**Comment by arunoda** [comment]
Cool. Will do a review tommorow morning for sure.

**Comment by arunoda** [comment]
Okay. This looks great.
But somehow, CMD + SHIFT + B is not working but CTRL + SHIFT + B is working.
Once you fixed it, I think we could merge it.

Note: Our UI is now kind a complex and it's with some adhoc design. I'm moving the codebase to [Mantra](https://github.com/kadirahq/mantra/) specification within next 2 days. After that we may need to rework some stuff. (Possibly more testing)

**Comment by ritz078** [comment]
If only CMD+SHIFT+B is not working then maybe some other software that is open uses that shortcut. The solution will be to change the shortcut but anyways no shortcut is safe.

**Comment by arunoda** [comment]
I'm using Chrome on Mac.

**Comment by ritz078** [comment]
I have checked and its working fine on chrome. Definitely, some other app is using that shortcut on app level.

**Comment by arunoda** [comment]
That's fine. Let's keep it like this for now.

**Comment by arunoda** [comment]
I'll take this.

**Comment by arunoda** [comment]
Published this as: :`v1.18.0`

**Comment by arunoda** [comment]
Hi @ritz078,

I re-wrote the whole storybook again for a modular release. (did with `v1.20.0)
So, I couldn't get your code for fuzzy-search for that.
I had to do a release soon. (I don't like to ask other's PR to wait for this refactor)

I'm planning to do it by next-week. If you've some time I'm looking for your help.
Here's a doc about new [Storybook Internals](https://github.com/kadirahq/react-storybook/blob/master/docs/how_storybook_works.md).

**Comment by ritz078** [comment]
No problem. I am developing a separate fuzzy search component named [react-fuzzy-search](https://github.com/ritz078/react-fuzzy-search) . I will integrate it once the tests for that are completed.

**Comment by arunoda** [comment]
@ritz078 that'sweet.

**Pull request merged**